scripts/makefile_help.rb

```diff
@@ -0,0 +1,40 @@
+# Script that parses description strings (## help) out of the Makefile
+# and also uses the `make --print-database` output to expand out targets based on variables
+
+expanded_makefile = `make --dry-run --print-data-base --no-builtin-rules`
+
+# Map of lineno => rule name run by printing the Makefile database
+expanded_targets = expanded_makefile.scan(/\n\n(?<target>[^:]+):( (?<sources>[^\n]+))?
+((#  [^\n]+)?\n?)*?
+#  commands to execute \(from `Makefile', line (?<lineno>\d+)\):/m).
+  map { |target, *, lineno| [lineno.to_i, target] }.
+  to_h
```

Comment:
I see Rubocop isn't taking nicely to this.

```
scripts/makefile_help.rb:7:43: C: [Correctable] Layout/FirstMethodArgumentLineBreak: Add a line break before the first argument of a multi-line method argument list.
expanded_targets = expanded_makefile.scan(/\n\n(?<target>[^:]+):( (?<sources>[^\n]+))? ...
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
scripts/makefile_help.rb:7:43: W: Lint/MixedRegexpCaptureTypes: Do not mix named captures and numbered captures in a Regexp literal.
expanded_targets = expanded_makefile.scan(/\n\n(?<target>[^:]+):( (?<sources>[^\n]+))? ...
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
45bb2ab3d8e806974b5783efe22f12186bac8005

added some ignores, because this is `/m` but not `/x` regex so whitespace matters and indenting the regex causes problems

Also mixing capture groups...these helped me remember what to grab, so I am just disabling for now